ADR/0044-spdx-support.md

```diff
@@ -0,0 +1,162 @@
+
+| CycloneDX Attribute          | SPDX Attribute                                                |
+|------------------------------|---------------------------------------------------------------|
+| component.purl = `<PURL>`    | package.externalRefs = [{referenceCategory:”PACKAGE-MANAGER”, |
+|                              |                          referenceType:purl,                  |
+|                              |                          referenceLocator: `<PURL>`           |
+|                              |                          }]                                   |
+
+
+#### Component.properties
+CycloneDX components properties describe mapping of string:string properties for given component. SPDX component doesn’t have anything similar to cyclonedx properties. SPDX Package annotations are the only attribute where custom data can be stored and the only “customizable” field where there is comment which is a simple string. Due to that fact, cycloneDX property in format of {“name”: <string>, “value”: <string>} is encoded into json string. There can be also annotations produced by other tools. Therefore to be able to tell annotation comment is json encoded, annotator should ends with string “:jsonencoded”
+
+| CycloneDX Attribute                       | SPDX Attribute                              |
+|-------------------------------------------|---------------------------------------------|
+| components.properties = [                 | package.annotations = [                     |
+|   {“name”: …, “value”: …}                 |   {..., annotator: "`<tool>`:jsonencoded”     |
+| ]                                         | ]                                           |
+
+
+#### Formulations
+CycloneDX formulations describe how the container was manufactured. In SPDX, Relationship elements can be used for the same purpose. All elements in SPDX have SPDXID attribute which is an element identifier unique in the whole SBOM document. Relationship element describes relation between two elements using their SPDXID and relationship type. Relationship type BUILD_TOOL_OF can be used to express the relationship of packages which were used to build the container.
+
+| CycloneDX Attribute             | SPDX Attribute                                             |
+|---------------------------------|------------------------------------------------------------|
+| Formulations.components = [{}]  | Relationships = [                                          |
+|                                 |     {                                                      |
+|                                 |          spdxElementId = `<ROOT-DOCUMENT-ID>`,             |
+|                                 |          relationshipType=DESCRIBES,                       |
+|                                 |          relatedSpdxElement=`<CONTAINER-IMAGE-ID>`,        |
+|                                 |     },                                                     |
```

Comment:
I don't understand why we'd need to add an additional `DESCRIBES` relationship for the builder image. It's connected to the `DESCRIBE`d image transitively by the `BUILD_TOOL_OF` relationship already isn't it? Here's a simple example (in tag form) of a similar situation: only the "thing" has a `DESCRIBES` relationship, not the other things which were used to build it or which it was build from: https://github.com/spdx/spdx-examples/blob/master/software/example1/spdx2.2/example1.spdx

Reply:
DESCRIBES relations is for container image of image which is being processed, not for builder image.

Reply:
Cachi2 doesn't use metadata.component and for cdx format, builder deps are populated into formulations. I think metadata.components shouldn't even be mentioned here as we don't work with it explicitly.

Reply:
Now that we have a section for `metadata.component`, could this go there?

Reply:
But this is not related to metadata.component. Metadata.component is translated as new root package of spdx. Formulations.components are translated as packages with BUILD_TOOL_OF relationships to the root package

Reply:
This hunk is for `ROOT-DOCUMENT-ID DESCRIBES CONTAINER-IMAGE-ID`. That's the `metadata.component` no?

Reply:
Ah, I see. But that's mentioned there only for completeness

Reply:
So what do you want to do with this? Remove it or move to correct section?

Reply:
I can remove it if you think it's rather confusing than informative